### PR TITLE
Switchover from Vercel to GitHub Pages

### PR DESCRIPTION
**Pull request by rezrah**
Points certain primer.style/* routes to new GitHub Pages origins via vercel-based url rewrites.

### Reviewers
During the migration from Vercel to GitHub Pages, certain pieces of content and functionality _may_ not have ported over correctly.

We would appreciate if maintainers of each repo can assist with a smoke test of their respective documentation sites, as they would be most familiar with how it functions at present.

1. Please open the URL for the docs you're familiar with below
2. Notify us via comment if you encounter any noticeable regressions to content, links or assets. 
3. If everything looks good, please give that item a check ✅  so we know it's good to go.

- [x] **CSS**: https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/css
- [x] **Primitives**: https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/primitives
- [x] **React**: https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/react
- [x] **React Brand**: https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/react-brand
- [x] **View Components**: https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/view-components
- [x] **Design**: https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/design
- [x] **Desktop**: https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/desktop
- [x] **Mobile**: https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/mobile
- [x] **CLI**: https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/cli
- [x] **Contribute**: https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/contribute
- [x] **Doctocat**: https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/doctocat

### CODE REVIEW

File: vercel.json [hunk 1]
```diff
@@ -1,23 +1,22 @@
...
+    {"source": "/primitives(/.*)?", "destination": "https://primer.github.io/primitives$1"},
+    {"source": "/mobile(/.*)?", "destination": "https://primer.github.io/mobile$1"},
+    {"source": "/react-brand(/.*)?", "destination": "https://primer.github.io/react-brand$1"},
     {
       "source": "/view-components/stories(/.*)?",
       "destination": "https://view-components-storybook.eastus.cloudapp.azure.com/view-components/stories$1"
     },
-    {"source": "/view-components(/.*)?", "destination": "https://view-components.vercel.app/view-components$1"},
-    {"source": "/desktop(/.*)?", "destination": "https://desktop-nu.vercel.app/desktop$1"},
+    {"source": "/view-components(/.*)?", "destination": "https://primer.github.io/view_components$1"},
```
Comment (rezrah) [inline vercel.json:18]: Known issue with this one ☝️. Fix [landing soon](https://github.com/primer/view_components/pull/1129).

**Comment by yaili** [comment]
primer.style/cli, primer.style/desktop, and primer.style/mobile:
- if you compare the new pages ([example](https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/cli/foundations/typography)) with the live pages ([same example](https://primer.style/cli/foundations/typography)) some styling isn't the same - navigation, h1 styling, breadcrumb style - is this because the current live sites are out of date and the new version is more current?

primer.style/cli:
- also noticed that some pages don't show the ToC on the right (for example all pages under [Foundations](https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/cli/foundations), compare with live site, where _some_ pages under [Foundations](https://primer.style/cli/foundations/typography) have a ToC)

primer.style/design:
- are we going to lose the contributors section, or is it not showing just because it's a demo link?
Screenshot 2022-04-21 at 15 11 14

**Comment by rezrah** [comment]
> * if you compare the new pages ([example](https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/cli/foundations/typography)) with the live pages ([same example](https://primer.style/cli/foundations/typography)) some styling isn't the same - navigation, h1 styling, breadcrumb style - is this because the current live sites are out of date and the new version is more current?

Nice catch. This is because the Doctocat version was changed. Upgrading that should fix these issues. Can this wait or is it a showstopper?

> are we going to lose the contributors section, or is it not showing just because it's a demo link?

I think this is because the GH API isn't being called. I'll investigate. I can mop this up after if that works for you @yaili?

**Comment by mperrotti** [comment]
Reviewed remaining items, and everything besides [ViewComponents](https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/view-components) seems to be working as expected.

I just want to confirm that we expect to see updated content about deploying to GitHub Pages in Doctocat. Example: https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/doctocat/getting-started

**Comment by rezrah** [comment]
> I just want to confirm that we expect to see updated content about deploying to GitHub Pages in Doctocat. Example: https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/doctocat/getting-started

Thanks fore reviewing these @mperrotti. The content you mentioned is captured in the main epic, but thanks for bringing it up. I'll be following up on it after this PR gets merged and we're confident that Pages is stable in production.

File: vercel.json (hunk 1 above)
Comment (rezrah) [reply vercel.json:18]: update: fixed https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/view-components

**Review by lesliecdubs — APPROVED** [review]
I am admittedly less familiar with the docs than other folks, but I did a pass across all of these and they LGTM.

One non-blocking thing I noticed in the React docs for specific components, e.g. https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/react/ActionList - the links to Storybook don't point directly to the relevant component story. This works as expected in the PVC docs. Is this something worth addressing later?

**Comment by rezrah** [comment]
> I am admittedly less familiar with the docs than other folks, but I did a pass across all of these and they LGTM.
> 
> One non-blocking thing I noticed in the React docs for specific components, e.g. https://primer-style-git-rezrah-pages-switchover-primer.vercel.app/react/ActionList - the links to Storybook don't point directly to the relevant component story. This works as expected in the PVC docs. Is this something worth addressing later?

The broken Storybook links in PRC are an existing issue, even on the Vercel deployed docs. URL is seemingly correct, so I'm not sure why it doesn't work without doing some further investigation. 

https://primer.style/react/ActionList 

1. Opening the Storybook link **in a new tab works** ✅ 
2. Opening the Storybook link **in the current tab doesn't work** ❌ 

I can create a bug report for this before we merge this PR, so it's clear this isn't a regression.

**Pull request merged**